### PR TITLE
refactor setup_dev_env script

### PR DESCRIPTION
**Pull request by edgarriba**
### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: setup_dev_env.sh [hunk 1]
```diff
@@ -36,11 +36,25 @@ fi
 # define a python version to initialise the conda environment.
 # by default we assume python 3.7.
 python_version=${PYTHON_VERSION:-"3.7"}
+pytorch_version=${PYTORCH_VERSION:-"1.6.0"}
+pytorch_mode=${PYTORCH_MODE:-""}  # use `cpuonly` for CPU or leave it in blank for GPU
```
Comment (shijianjian) [inline setup_dev_env.sh:40]: Would this auto-detect the device?

Probably something like:
```python
try:
    nvidia-smi
    pytorch+cuda
except:
    pytorch+cpu
```

File: setup_dev_env.sh [hunk 2]
```diff
@@ -36,11 +36,25 @@ fi
...
+source $conda_bin_dir/activate $dev_env_dir/envs/venv
+
+# install pytorch and torchvision
+conda install pytorch=$pytorch_version torchvision $pytorch_mode -c pytorch
+
+# install testing dependencies
+pip install -r requirements-dev.txt
+
+# install documentation dependencies
+pip install -r docs/requirements.txt
```
Comment (shijianjian) [inline setup_dev_env.sh:58]: Forgot adding in the file?

Comment (edgarriba) [reply setup_dev_env.sh:58]: Nope, this file has existed for a while used only by read the docs

File: setup_dev_env.sh (hunk 1 above)
Comment (edgarriba) [reply setup_dev_env.sh:40]: It's better to specify by hand in case we want to test cpuonly
